### ISSUE TITLE
Uncaught TypeError: Invalid Point: (NaN, NaN) - error parsing const vs let

### ISSUE THREAD

**Issue by sheeley** [gandm/language-babel#112]
This coincides with a transpiler error - it barfs on the `name` const for some reason:
```js
export default function(options) {
  function* createOrUpdate(next) {
    const name = this.request.body.fields.name
    console.log(name)
  }
  return createOrUpdate
}
```

```
"name" is read-only (This is an error on an internal node. Probably an internal error. Location has been estimated.)
```

Swapping to `let` from `const` causes error to not be thrown:
```js
export default function(options) {
  function* createOrUpdate(next) {
    let name = this.request.body.fields.name
    console.log(name)
  }
  return createOrUpdate
}
```

**Atom Version**: 1.2.4
**System**: Mac OS X 10.11.1
**Thrown From**: [language-babel](https://github.com/gandm/language-babel) package, v2.5.0

### Stack Trace

Uncaught TypeError: Invalid Point: (NaN, NaN)

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/point.js:65

TypeError: Invalid Point: (NaN, NaN)
    at Function.module.exports.Point.assertValid (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/point.js:65:15)
    at TextBuffer.module.exports.TextBuffer.clipPosition (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:980:13)
    at MarkerStore.module.exports.MarkerStore.setMarkerRange (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker-store.js:318:29)
    at Marker.module.exports.Marker.update (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:325:20)
    at Marker.module.exports.Marker.setHeadPosition (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:143:19)
    at Marker.module.exports.Marker.setHeadBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:173:32)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:80:31
    at Cursor.module.exports.Cursor.changePosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:674:7)
    at Cursor.module.exports.Cursor.setBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:78:19)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1336:23
    at TextEditor.module.exports.TextEditor.moveCursors (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1573:9)
    at TextEditor.module.exports.TextEditor.setCursorBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1335:19)
    at /Users/jsheeley/.atom/packages/language-babel/lib/transpiler.coffee:100:26
    at /Applications/Atom.app/Contents/Resources/app.asar/src/task.js:136:25
    at /Applications/Atom.app/Contents/Resources/app.asar/src/task.js:126:25
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at ChildProcess.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/task.js:78:34)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at handleMessage (internal/child_process.js:685:10)
    at Pipe.channel.onread (internal/child_process.js:440:11)
```

### Commands

```
     -0:47 editor:select-to-end-of-line (atom-text-editor.editor.is-focused)
     -0:46.7.0 core:copy (atom-text-editor.editor.is-focused)
     -0:46.5.0 core:paste (atom-text-editor.editor.is-focused)
     -0:46.2.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:46 core:paste (atom-text-editor.editor.is-focused)
     -0:45.5.0 editor:move-to-first-character-of-line (atom-text-editor.editor.is-focused)
     -0:44.4.0 core:save (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:33.6.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:33.2.0 core:save (atom-text-editor.editor.is-focused)
  3x -0:26.4.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:25.1.0 core:save (atom-text-editor.editor.is-focused.autocomplete-active)
  2x -0:18.7.0 editor:toggle-line-comments (atom-text-editor.editor.is-focused)
     -0:15.7.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:12.8.0 core:move-down (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:12.6.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
  2x -0:12.1.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "themes": [
      "solarized-seti-ui",
      "monokai-seti"
    ],
    "ignoredNames": [
      "node_modules/*",
      "bower_components/*"
    ],
    "disabledPackages": [
      "jscs-fixer",
      "react",
      "standard-formatter",
      "linter-js-standard"
    ]
  },
  "language-babel": {
    "disableWhenNoBabelrcFileInPath": true,
    "babelTranspilePath": "dist",
    "transpileOnSave": true
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.3.0
atom-minify, v0.6.2
auto-detect-indentation, v0.4.2
auto-indent, v0.5.0
autocomplete-paths, v1.0.2
autocomplete-python, v0.10.4
close-after-last-tab, v0.1.6
highlight-selected, v0.11.1
jsonlint, v1.1.2
language-babel, v2.5.0
language-gitignore, v0.2.0
language-gradle, v0.0.3
language-terraform, v0.6.0
linter, v1.11.2
linter-coffeescript, v1.0.0
linter-eslint, v5.2.4
linter-jsonlint, v1.0.1
monokai-seti, v0.7.0
package-manager-commands, v2.0.3
pigments, v0.19.0
pretty-json, v0.4.1
set-syntax, v0.3.0
solarized-seti-ui, v0.8.6
sync-settings, v0.6.0
synced-sidebar, v0.4.3
tag, v0.3.0

# Dev
No dev packages
```

**Comment by gandm** [comment]
Thanks for the report, I'll fix this. Because the bad const  babel error doesn't return a line/column number for the error I can't position the cursor in the TextBuffer. Not sure why Babel doesn't return a line/col fort this error but I should have trapped this eventuality.

**Comment by gandm** [comment]
Fixed in 2.5.1. Thanks.

**Comment by sheeley** [comment]
@gandm: thanks! that was fast.
